documentation/high-level-introduction.md

```diff
@@ -0,0 +1,257 @@
+`<Import Project="$(MSBuildExtensionsPath)\Xamarin\Mac\Xamarin.Mac.CSharp.targets" />`
+
+- Solution files (.sln) combine one or more projects for editing and building. They contain some build dependency and configuration information, but project files are the main level of build knowledge.
+- Target Files (.targets)
+  - These files are generally provided by the SDKs
+  - These files generally describe a set of steps (targets) which need to be executed to process the build
+  - These tend to contain many UsingTask items to reference task assembly targets and Target to define the targets and dependencies
+- Property file (.props)
+  - These files are generally provided by the SDKs
+  - These files generally set a number of properties used by the targets
+  - These tend to contain many PropertyGroup and ItemGroup items defining values needed in the builds.
+
+All of these are are [XML Structured](#XML-Structure) text files.
+
+As I noted above, there is nothing truly special with the names and suffixes, they are just standard conventions you should follow.
+
+### Libraries 
+
+Many parts of the build involve invoking outside tools or operations that would be burdensome to write in XML. Task assemblies allow extension of the msbuild by defining types in a C# library that can be called by msbuild targets.
+
+
+- Task Assemblies (.dll)
+  - These files are generally provided by the SDKs
+  - These are C# libraries which reference various msbuild libraries
+  - These define [tasks](#MSBuild-Tasks) which can be invoked inside defined targets to carry out parts of the builds.
+  - They are loaded when other msbuild files use `UsingTask` nodes
+
+
+
+## Data
+
+
+- There is only one data type in msbuild for values. Everything is a string.
+- msbuild distinguished between two types of data, properties and items:
+  - Properties are single named elements 
+    - Think of accessing into a Dictionary<string, string> with the name type being the key
+    - They are defined in PropertyGroups
+
+            <PropertyGroup>
+              <DebugSymbols>true</DebugSymbols>
+            </PropertyGroup>
+
+       - This defines a property "DebugSymbols” with the value “true”
+    - They are referenced later via $(NAME)
+  - Items are a list of properties containing one or more elements
+    - Think of accessing into a Dictionary<string, List<string>> with the name type being the key
+    - They are defined in ItemGroups
+
+            <ItemGroup>
+              <Reference Include="System" />
+            </ItemGroup>
+
+      - This includes the value “System” in the Item Reference, which may already have zero or more items
+    - There are also Exclude, Remove, and other actions you can do on an Item
+    - They are referenced later via @(LIST_NAME) or [list transforms](#List-Transforms).
+    - The may be represented by a semicolon separated string of items
+- **Variables will return empty string if not defined!**
```

Comment:
maybe add something about not throwing an error? It's implied, but it would be nice to be explicit.